src/django/api/models/facility_download_limit.py

```diff
@@ -0,0 +1,91 @@
+from __future__ import annotations
+import uuid
+from typing import Optional
+from django.db import models
+from django.utils import timezone
+from django.db import transaction
+from api.constants import FacilitiesDownloadSettings
+
+from api.models.facility_download_limit_manager import (
+    FacilityDownloadLimitManager
+)
+
+
+class FacilityDownloadLimit(models.Model):
+    """
+    Stores the number of facility records allowed for free download per
+    calendar year, the number of paid facility records for non-API
+    users, the date of last update of free facilities records, and the
+    date when paid facility records were purchased.
+    """
+    uuid = models.UUIDField(
+        null=False,
+        default=uuid.uuid4,
+        unique=True,
+        editable=False,
+        help_text='Unique identifier for the facility download limit record.'
+    )
+    user = models.OneToOneField(
+        'User',
+        null=False,
+        on_delete=models.CASCADE,
+        help_text='The user to whom the download limit applies.'
+    )
+    free_download_records = models.PositiveIntegerField(
+        null=False,
+        blank=False,
+        default=FacilitiesDownloadSettings.FACILITIES_DOWNLOAD_LIMIT,
+        help_text=('The number of facilities the user '
+                   'can download per calendar year for free.')
+    )
+    paid_download_records = models.PositiveIntegerField(
+        null=False,
+        blank=False,
```

Comment:
- `null` is `False` by default. See the docs -> https://docs.djangoproject.com/en/3.2/ref/models/fields/#django.db.models.Field.null
- `blank` is `False` by default. See the docs -> https://docs.djangoproject.com/en/3.2/ref/models/fields/#django.db.models.Field.blank